src/component/Header/Header.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface HeaderProps {
+  todos: Todo[];
+  addTodo: (title: string) => void;
+  query: string;
+  setQuery: (query: string) => void;
+  toggleAll: boolean;
+  setToggleAll: (value: boolean) => void;
+}
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  addTodo,
+  query,
+  setQuery,
+  toggleAll,
+  setToggleAll,
```

Comment:
Consider renaming 'setToggleAll' to 'onToggleAll' to better represent an event handler.